File: Ahorn/entities/lantern.jl
module JungleHelperLantern

using ..Ahorn, Maple

@mapdef Entity "JungleHelper/Lantern" Lantern(x::Integer, y::Integer, sprite::String="", onlyIfMaddyNotHolding::Bool=false)

const placements = Ahorn.PlacementDict(
    "Lantern (Jungle Helper)" => Ahorn.EntityPlacement(
        Lantern
    )
)

sprite = "JungleHelper/Lantern/LanternEntity/lantern_00"

function Ahorn.selection(entity::Lantern)
    x, y = Ahorn.position(entity)
    return Ahorn.getSpriteRectangle(sprite, x, y + 5)
end

Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::Lantern, room::Maple.Room) = Ahorn.drawSprite(ctx, sprite, 0, 5)

end

File: Loenn/entities/lantern.lua
local utils = require("utils")

local lantern = {}
lantern.name = "JungleHelper/Lantern"
lantern.depth = 0
lantern.placements = {
    {
        name = "default",
        data = {
            sprite = "",
            lightSprite = "JungleHelper/Lantern/Overlay",
            onlyIfMaddyNotHolding = false
        }
    }
}

lantern.offset = { 10, 5 }
lantern.justification = { 0, 0 }
lantern.texture = "JungleHelper/Lantern/LanternEntity/lantern_00"

lantern.fieldOrder = {
    "x","y","sprite","lightSprite","onlyIfMaddyNotHolding"
}

function lantern.selection(room, entity)
    return utils.rectangle(entity.x - 5, entity.y - 2, 10, 10)
end

return lantern
